qiskit/algorithms/gradients/base_estimator_gradient.py

```diff
@@ -0,0 +1,122 @@
+        observables: Sequence[BaseOperator | PauliSumOp],
+        parameter_values: Sequence[Sequence[float]],
+        partial: Sequence[Sequence[Parameter]] | None = None,
+        **run_options,
+    ) -> EstimatorGradientResult:
+        """Run the job of the gradients of expectation values.
+
+        Args:
+            circuits: The list of quantum circuits to compute the gradients.
+            observables: The list of observables.
+            parameter_values: The list of parameter values to be bound to the circuit.
+            partial: The list of Parameters to calculate only the gradients of the specified parameters.
+                Defaults to None, which means that the gradients of all parameters will be calculated.
+            run_options: Backend runtime options used for circuit execution.
+
+        Returns:
+            The job object of the gradients of the expectation values. The i-th result corresponds to
+            ``circuits[i]`` evaluated with parameters bound as ``parameter_values[i]``.
+
+        Raises:
+            QiskitError: Invalid arguments are given.
+        """
+        # Validation
+        if len(circuits) != len(parameter_values):
+            raise QiskitError(
+                f"The number of circuits ({len(circuits)}) does not match "
+                f"the number of parameter value sets ({len(parameter_values)})."
+            )
+
+        if len(circuits) != len(observables):
+            raise QiskitError(
+                f"The number of circuits ({len(circuits)}) does not match "
+                f"the number of observables ({len(partial)})."
+            )
+
+        if partial is not None:
+            if len(circuits) != len(partial):
+                raise QiskitError(
+                    f"The number of circuits ({len(circuits)}) does not match "
+                    f"the number of partial parameter sets ({len(partial)})."
+                )
+
+        for i, (circuit, parameter_value) in enumerate(zip(circuits, parameter_values)):
+            if len(parameter_value) != circuit.num_parameters:
+                raise QiskitError(
+                    f"The number of values ({len(parameter_value)}) does not match "
+                    f"the number of parameters ({circuit.num_parameters}) for the {i}-th circuit."
+                )
+
+        for i, (circuit, observable) in enumerate(zip(circuits, observables)):
+            if circuit.num_qubits != observable.num_qubits:
+                raise QiskitError(
+                    f"The number of qubits of the {i}-th circuit ({circuit.num_qubits}) does "
+                    f"not match the number of qubits of the {i}-th observable "
+                    f"({observable.num_qubits})."
+                )
+
+        # The priority of run option is as follows:
+        # run_options in `run` method > gradient's default run_options > primitive's default setting.
+        run_options = run_options or self._default_run_options
```

Comment:
It would be better to update this on an element-by-element, rather than chossing dict. 
My PR may be helpful.
https://github.com/Qiskit/qiskit-terra/pull/8513/files#diff-c5bbeb4ba0adbcae7deae85bcb381895d135b9437bc9c30340e6ca12e8395b6dR528
(※ You don't need to use Option.)

Reply:
It would be nice to document this ordering.